docs/whatsnew/5.0.rst

```diff
@@ -36,3 +36,18 @@ The minimum required versions of these optional dependencies has also been updat
 The ``sunpy.database`` module is no longer actively maintained and has a number of outstanding issues.
 It has now been formally deprecated, and will be removed in sunpy 6.0.
 If you are using sunpy.database and would like to see a replacement, please join the discussion thread at https://community.openastronomy.org/t/deprecating-sunpy-database/495.
+
+Deprecation of ``sunpy.io.jp2``
+================================
+The ``sunpy.io.jp2`` module is deprecated, and will be removed in version 5.1.
+This because it was designed for internal use only, and removing it from the public API gives the developers more flexibility to modify it without impacting users.
+
+Deprecation of ``sunpy.io.file_tools``
+================================
+The ``sunpy.io.file_tools`` module is deprecated, and will be removed in version 5.1.
+This because it was designed for internal use only, and removing it from the public API gives the developers more flexibility to modify it without impacting users.
+
+Deprecation of ``sunpy.io.cdf``
+================================
+The ``sunpy.io.cdf`` module is deprecated, and will be removed in version 5.1.
+This because it was designed for internal use only, and removing it from the public API gives the developers more flexibility to modify it without impacting users.
```

Comment:
```suggestion
Deprecation of IO readers``sunpy.io.jp2``
=========================================
``sunpy.io.cdf``, ``sunpy.io.file_tools`` and ``sunpy.io.jp2`` sub-modules have been depreciated, and will be removed in version 5.1.

This because they are designed for internal use only, and removing it from the public API gives the developers more flexibility to modify it without impacting users.
```